terra_layer/models.py

```diff
@@ -68,6 +70,17 @@ class Meta:
             ('can_manage_layers', 'Can manage layers'),
         )
 
+    def save(self, **kwargs):
+        super().save(**kwargs)
+
+        # Find the upmost group for this layer
+        group = self.group
+        if group:
+            while group.parent:
```

Comment:
You don't need to go up in the group tree, you can you group.view

Reply:
What if tree is nested? Like on https://terralego.netlify.com/?path=/story/modules-visualizer-layerstree--component

Reply:
Every group has a view attribute, that in which view it appears, event children has this view attribute.

Reply:
:+1: